### PR TITLE
fix try to copy null will cause crash on iOS

### PR DESCRIPTION
**Pull request by KagurazakaHanabi**
Closes flutter/flutter#30850

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm [hunk 1]
```diff
@@ -224,7 +224,11 @@ - (NSDictionary*)getClipboardData:(NSString*)format {
 
 - (void)setClipboardData:(NSDictionary*)data {
   UIPasteboard* pasteboard = [UIPasteboard generalPasteboard];
-  pasteboard.string = data[@"text"];
+  if (data[@"text"]) {
+    pasteboard.string = data[@"text"];
+  } else {
+    pasteboard.string = @"null";
```
Comment (GaryQian) [inline shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm]: Passing an empty string may be better than "null" here. If the contents were actually the string "null" we would not be able to distinguish between an incorrect usage of the setClipboardData API vs the actual content. End users would be very surprised to see the term "null" if this was accidentally used improperly.

**Comment by GaryQian** [comment]
Could you explain what use case leads to passing a nil value under data["text"]?

**Comment by KagurazakaHanabi** [comment]
If there is a problem with the code, maybe will passing nil to data["text"], which will cause a crash on iOS, copy "null" on Android
This is not a serious error, so shouldn't crash

**Comment by GaryQian** [comment]
Ahh I see. Thanks for the quick changes! Since We currently pass "null" on android, I think it is acceptable to just make iOS do the same thing for now. Changing it to "" would be a very minor breaking change, and it is best to avoid that when possible. Do you mind changing it back? Thanks!

After that, LGTM!

**Comment by KagurazakaHanabi** [comment]
Done

**Pull request merged**